core/trino-main/src/main/java/io/trino/security/AccessControl.java

```diff
@@ -313,7 +314,7 @@ public interface AccessControl
      *
      * @throws AccessDeniedException if not allowed
      */
-    void checkCanCreateView(SecurityContext context, QualifiedObjectName viewName);
+    void checkCanCreateView(SecurityContext context, QualifiedObjectName viewName, Optional<ViewSecurity> security);
```

Comment:
It is not needed. This method is simply not called for INVOKER.

Reply:
Uhh interesting, thanks for the pointer!
Did I get it right that users can always create views (with invoker), and there is no way to prohibit that?

Reply:
Oh good point. I was thinking about `io.trino.security.ViewAccessControl#checkCanSelectFromColumns` that is checking this during the query of the view.

You are right, we are checking this for view creation disregarding security mode.

Reply:
I don't think this should be optional.

Also why we want access control grant/revoke access to create a view depending if it is INVOKER or DEFINER?